doc/manual.rst

```diff
@@ -5578,7 +5576,35 @@ types that will match the typedesc param:
 
 The constraint can be a concrete type or a type class.
 
+type operator
```

Comment:
I really think this should be named ``typeof``. With ``type`` now being a typeclass it's just too confusing otherwise.

Reply:
The point is that this not an operator anymore. It's a type coercion (you are constructing a type value from an expression). This enables you to specify constrains on the type being constructed, arguably not a particularly important feature, but still give me the feeling of things clicking well together.

Reply:
Just for curiosity: what do you mean by "This enables you to specify constrains on the type being constructed"?

Reply:
You can say `type[Enumerable](foo())`, which will get the return type of `foo` while ensuring that it matches the `Enumerable` concept.

Reply:
You don't "coerce" a value into a type though. But it's nitpicking.